services/navigation/builtin/builtin.go

```diff
@@ -823,25 +823,20 @@ func (svc *builtIn) Paths(ctx context.Context, extra map[string]interface{}) ([]
 		return nil, err
 	}
 
-	geoPoints := make([]*geo.Point, 0, len(ph[0].Plan.Steps))
-	for _, s := range ph[0].Plan.Steps {
-		if len(s) > 1 {
-			return nil, errors.New("multi component paths are unsupported")
-		}
-		var pose spatialmath.Pose
-		for _, p := range s {
-			pose = p
-		}
-
-		geoPoint := geo.NewPoint(pose.Point().Y, pose.Point().X)
-		geoPoints = append(geoPoints, geoPoint)
+	path := ph[0].Plan.Path()
+	geoPoints := make([]*geo.Point, 0, len(path))
+	poses, err := path.GetFramePoses(svc.base.Name().ShortName())
+	if err != nil {
+		return nil, err
 	}
-
-	path, err := navigation.NewPath(ewp.waypoint.ID, geoPoints)
+	for _, p := range poses {
+		geoPoints = append(geoPoints, geo.NewPoint(p.Point().Y, p.Point().X))
+	}
+	navPath, err := navigation.NewPath(ewp.waypoint.ID, geoPoints)
```

Comment:
Does this not do the wrong thing if there is more than one component in the plan?

Reply:
I'm sorry I don't follow this question

Reply:
The old code checked if any of the steps had more than one component in them b/c if they do then it means we would be adding waypoints for each component which is not what the caller would expect.

I see that this code doesn't do any of that checking.

Is there anything preventing this function from returning the wrong results if there is more than one component in the plan?